src/services/CurrencyService.ts

```diff
@@ -43,39 +41,30 @@ export function useFetchHistoryData(selectedCurrencies: CurrencyHistoryData[], p
     const [data, setData] = useState<HistoryData[]>([])
     const [loading, setLoading] = useState<boolean>(true)
 
-    const request = (url: string) => from(fetch(url)
-        .then(data => data.json()));
-
-    const checkIfPolishCurrencySelected = selectedCurrencies.findIndex(currency =>  currency.code === polishCurrencyCode)
+    const checkIfPolishCurrencySelected = selectedCurrencies.findIndex(currency => currency.code === polishCurrencyCode)
 
     useEffect(() => {
-
-        console.log('useEffect')
-
-        if (checkIfPolishCurrencySelected) {
-            selectedCurrencies = selectedCurrencies.filter(selectedCurrency => {
-                return selectedCurrency.code !== polishCurrencyCode
-            })
-        }
-
-        console.log(selectedCurrencies)
-
-        if (selectedCurrencies.length === 2) {
+        if (checkIfPolishCurrencySelected === -1) {
             forkJoin({
                 firstCurrency: request(`http://api.nbp.pl/api/exchangerates/rates/${selectedCurrencies[0]?.table}/${selectedCurrencies[0]?.code}/last/${period}`),
                 secondCurrency: request(`http://api.nbp.pl/api/exchangerates/rates/${selectedCurrencies[1]?.table}/${selectedCurrencies[1]?.code}/last/${period}`)
             }).subscribe(({firstCurrency, secondCurrency}) => {
-                setData(prepareHistoryData(firstCurrency.rates, secondCurrency.rates));
+                setData(prepareHistoryData(firstCurrency.rates, false, secondCurrency.rates));
                 setLoading(false)
 
             })
         }
 
-        if (selectedCurrencies.length === 1) {
+        if (checkIfPolishCurrencySelected > -1) {
+
+            selectedCurrencies = selectedCurrencies.filter(selectedCurrency => {
```

Comment:
if you have an index of `polishCurrency` then maybe we can use it and remove this element from existing array by using `splice` method instead of creating new array? :D 

can you check please something like this: 
```JavaScript
 selectedCurrencies.splice(checkIfPolishCurrencySelected, 1)
```

but then, we should consider changing name of the `checkIfPolishCurrencySelected` variable. Maybe `selectedPolishCurrencyIndex`?

Reply:
You are right as always Radzio madzio tadzio kadzio